kolena/detection/_internal/model.py

```diff
@@ -79,6 +81,7 @@ class BaseModel(ABC, Frozen, WithTelemetry):
 
     @validate_arguments(config=ValidatorConfig)
     def __init__(self, name: str, workflow: WorkflowType, metadata: Optional[Dict[str, Any]] = None):
+        validate_name(name, FieldNames.MODEL_NAME)
```

Comment:
I don't think this works well in Python 3.11 (which we support):

```
$ poetry run python3
Python 3.11.1 (main, Feb 13 2023, 09:21:24) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from kolena.detection import Model
>>> Model("")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gordonhart/Software/kolena/kolena/_utils/instrumentation.py", line 40, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gordonhart/Software/kolena/kolena/detection/model.py", line 56, in __init__
    super().__init__(name, WorkflowType.DETECTION, metadata)
  File "/Users/gordonhart/Software/kolena/kolena/_utils/instrumentation.py", line 40, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/decorator.py", line 40, in pydantic.decorator.validate_arguments.validate.wrapper_function
  File "pydantic/decorator.py", line 134, in pydantic.decorator.ValidatedFunction.call
  File "pydantic/decorator.py", line 206, in pydantic.decorator.ValidatedFunction.execute
  File "/Users/gordonhart/Software/kolena/kolena/detection/_internal/model.py", line 84, in __init__
    validate_name(name, FieldNames.MODEL_NAME)
  File "/Users/gordonhart/Software/kolena/kolena/_utils/validators.py", line 30, in validate_name
    raise ValueError(f"{field_name_str} must be non empty")
ValueError: FieldNames.MODEL_NAME must be non empty
```

Probably want to pass `FieldNames.MODEL_NAME.value` here or have `validate_name` take an `Optional[FieldNames]` and access the value itself.

Reply:
will update

Reply:
Will update and test with 3.11.

Reply:
Updated to use `Optional[FieldNames]`.

This is the new error on python 3.11:

```
Python 3.11.3 (main, Jul  4 2023, 22:05:40) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from kolena.detection import Model
>>> Model("")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/niteshsandal/Workspace/kolena/kolena/_utils/instrumentation.py", line 40, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/niteshsandal/Workspace/kolena/kolena/detection/model.py", line 56, in __init__
    super().__init__(name, WorkflowType.DETECTION, metadata)
  File "/Users/niteshsandal/Workspace/kolena/kolena/_utils/instrumentation.py", line 40, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/niteshsandal/Library/Caches/pypoetry/virtualenvs/kolena-1YW62IuT-py3.11/lib/python3.11/site-packages/pydantic/decorator.py", line 40, in wrapper_function
    return vd.call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/niteshsandal/Library/Caches/pypoetry/virtualenvs/kolena-1YW62IuT-py3.11/lib/python3.11/site-packages/pydantic/decorator.py", line 134, in call
    return self.execute(m)
           ^^^^^^^^^^^^^^^
  File "/Users/niteshsandal/Library/Caches/pypoetry/virtualenvs/kolena-1YW62IuT-py3.11/lib/python3.11/site-packages/pydantic/decorator.py", line 201, in execute
    return self.raw_function(**d, **var_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/niteshsandal/Workspace/kolena/kolena/detection/_internal/model.py", line 84, in __init__
    validate_name(name, FieldNames.MODEL_NAME)
  File "/Users/niteshsandal/Workspace/kolena/kolena/_utils/validators.py", line 32, in validate_name
    raise ValueError(f"{field_name_str} must be non empty")
ValueError: Model name must be non empty

```